docusaurus/docs/create-a-plugin/extend-a-plugin/add-router.md

```diff
@@ -0,0 +1,85 @@
+	queryTypeMux.HandleFunc("traces", ds.handleTracesQuery)
+	queryTypeMux.HandleFunc("", ds.handleQueryFallback)
+	ds.queryHandler := queryTypeMux
+	return ds
+}
+
+func (d *MyDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	return d.queryHandler.QueryData(ctx, req)
+}
+
+// handleMetricsQuery handle queries of query type "metrics".
+// All queries in backend.QueryDataRequest is guaranteed to only
+// include queries with queryType "metrics".
+func (d *MyDatasource) handleMetricsQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+
+// handleLogsQuery handle queries of query type "logs".
+// All queries in backend.QueryDataRequest is guaranteed to only
+// include queries with queryType "logs".
+func (d *MyDatasource) handleLogsQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+
+// handleTracesQuery handle queries of query type "logs".
+// All queries in backend.QueryDataRequest is guaranteed to only
+// include queries with queryType "traces".
+func (d *MyDatasource) handleTracesQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+
+// handleQueryFallback handle queries without a matching query type handler registered.
+func (d *MyDatasource) handleQueryFallback(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// implementation...
+}
+```
+
+## Advanced usage
+
+An example of using [`QueryTypeMux`](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/datasource#QueryTypeMux) can be found for Grafana's built-in Grafana Azure Monitor data source. Refer to this code for examples of implementation:
+
+- [grafana/azuremonitor.go](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L49) at 164ce63e280565cc960fb0e0d9f074ec003d5a2a - See the code at the [grafana/grafana](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L49) repo.
+- [grafana/azuremonitor.go](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L182-L201) at 164ce63e280565cc960fb0e0d9f074ec003d5a2a - See the code at the [grafana/grafana](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L182-L201) repo.
+- [grafana/azuremonitor.go](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L55-L57) at 164ce63e280565cc960fb0e0d9f074ec003d5a2a - See the code at the [grafana/grafana](https://github.com/grafana/grafana/blob/164ce63e280565cc960fb0e0d9f074ec003d5a2a/pkg/tsdb/azuremonitor/azuremonitor.go#L55-L57) repo.
```

Comment:
```suggestion
- [create query type multiplexer](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/testdata.go#L22) and [calls registerScenarios](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/testdata.go#L44)
- [registerScenarios method](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/scenarios.go#L33) uses a [helper method](https://github.com/grafana/grafana/blob/623ee3a2be5c4cd84c61b6bbe82a32d18cc29828/pkg/tsdb/grafana-testdata-datasource/scenarios.go#L204-L207) to register each query type handler. The latter also shows how you can wrap the actual handler in another handler to apply common functionality/middleware to all handlers, e.g. logging and traces. 
```

Reply:
@josmperez seems like you missed to incorporate this suggestion, but added my earlier suggestion?